module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp

```diff
@@ -765,6 +747,24 @@ namespace bsp::battery_charger
         return current;
     }
 
+    int getCurrentMeasurement()
+    {
+        auto value = fuelGaugeRead(Registers::Current_REG);
+        int current;
+        // 2's compliment into decimal
+        if (value.second & 0x8000) {
+            // negative numbers
+            std::bitset<16> currentBitset = std::bitset<16>(value.second - 1);
+            currentBitset.flip();
+            current = static_cast<int>((static_cast<std::uint16_t>(currentBitset.to_ulong()) * -1) * currentSenseGain);
```

Comment:
wouldn't it be more readable if it were moved to a separate function? Don't we already have one?

Reply:
We do not have an utility for it. I can extract it.